core/services/keeper/delegate.go

```diff
@@ -95,6 +95,14 @@ func (d *Delegate) ServicesForSpec(spec job.Job) (services []job.Service, err er
 		Logger:                   svcLogger,
 		SyncUpkeepQueueSize:      chain.Config().KeeperRegistrySyncUpkeepQueueSize(),
 	})
+
+	/* check that node has proper configuration for the given chain ID */
+	chainId := chain.Client().ChainID().Text(10)
+	if hasEIP1559Support(chainId) && !chain.Config().EvmEIP1559DynamicFees() {
```

Comment:
IMO don't bother checking it. Let the simulation fail instead and add a useful error message to the simulation fail that explains it.

In addition you might consider adding a check to the validation stage of when you add a job.